examples/basic_tutorial_BERT.ipynb

```diff
@@ -0,0 +1,1172 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c0b6eb14",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "trainloader = torch.utils.data.DataLoader(ds1, batch_size=batch_size,\n",
+    "                                          shuffle=True, num_workers=2)\n",
+    "testloader = torch.utils.data.DataLoader(ds2, batch_size=batch_size,\n",
+    "                                         shuffle=False, num_workers=2)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2533b495",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dl_builder2 = DataLoaderBuilder((FromArray(test, train_labels), FromArray(test2, validation_labels)))\n",
+    "dl_tr2, dl_val2, _ = dl_builder2.build(({\"batch_size\": 8}, {\"batch_size\": 8}))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "534a126d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class Net(nn.Module):\n",
+    "    def __init__(self):\n",
+    "        super().__init__()\n",
+    "        self.linear=nn.Linear(2*heads_to_keep*50*5,2)\n",
+    "\n",
+    "    def forward(self, x):\n",
+    "        x = torch.flatten(x, 1) \n",
+    "        x = self.linear(x)\n",
+    "        return x\n",
+    "\n",
+    "\n",
+    "net = Net()\n",
+    "\n",
+    "criterion = nn.CrossEntropyLoss()\n",
+    "optimizer = Adam(net.parameters(), lr=0.000067)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "421526f2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pipe2 = Trainer(net, (dl_tr2, dl_val2), criterion, writer)\n",
+    "# train the model with learning rate scheduler\n",
+    "pipe2.train(\n",
+    "    Adam,\n",
+    "    20,\n",
+    "    False,\n",
+    "    lr_scheduler=ExponentialLR,\n",
+    "    scheduler_params={\"gamma\": 0.9},\n",
+    "    profiling=False,\n",
+    "    store_grad_layer_hist=True,\n",
+    "    writer_tag=\"line\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "13bdfa4b",
+   "metadata": {},
+   "source": [
+    "## Conclusion"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2e8f464b",
+   "metadata": {},
+   "source": [
+    "In this tutorial, we used *giotto-deep* to\n",
+    "1. Fit an off the shelf Huggingface model;\n",
+    "2. Fine tune it;\n",
+    "3. Extract its inner workings to;\n",
+    "4. construct a pruned model.\n",
+    "\n",
+    "Depending on the choices made along the way, the pruned model may be anything from strong to terrible. To increase performance, you may\n",
+    "\n",
+    "1. Add more data (remember this notebook is heavily subsampled);\n",
+    "2. Spend more time fine-tuning the BERT; \n",
+    "3. Fit more complicated models;\n",
+    "4. Optimize the parameters.\n"
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python [conda env:torch] *",
+   "language": "python",
+   "name": "conda-env-torch-py"
```

Comment:
the CI is complaining about this name: 

```
jupyter_client.kernelspec.NoSuchKernel: No such kernel named conda-env-torch-py
```

Can you please change this to:
```
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
```

Reply:
Thank you! Should be fixed now

Reply:
@hkirvesl can you please use `os.path.join` to build file path please? windows and posix OS have a different convention (back and forward slashes)

<img width="1205" alt="Screen Shot 2022-09-29 at 19 56 54" src="https://user-images.githubusercontent.com/38946719/193107037-1447a39c-3506-4e18-baef-2a8131209c19.png">

Reply:
Thank you for the suggestion. Done.